clang/lib/Parse/ParseDeclCXX.cpp

```diff
@@ -1110,6 +1110,8 @@ Decl *Parser::ParseStaticAssertDeclaration(SourceLocation &DeclEnd) {
   }
 
   T.consumeClose();
+  if (T.getCloseLocation().isInvalid())
+    return nullptr;
```

Comment:
`consumeClose` returms true on failure so this can just be `if(T.consumeClose()) return nullptr`

Reply:
@cor3ntin Thanks for the review. I've added changes to omit redundant checks.